### PR TITLE
[Docs] fix kafka format doc

### LINKED ISSUE #6632: [Bug] [Kafka] Kafka format debezium-json is not supported
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

use kafka connector debezium-json format 

### SeaTunnel Version

2.3.4

### SeaTunnel Config

```conf
"sink": [
    {
      "plugin_name": "kafka",
      "format": "debezium-json",
...
    }
  ]
```

### Running Command

```shell
rest api submit job
```

### Error Exception

```log
{
    "status": "fail",
    "message": "ErrorCode:[API-06], ErrorDescription:[Factory initialize failed] - Unable to create a sink for identifier 'kafka'."
}
```

### Zeta or Flink or Spark Version

zeta

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liunaijie**
### Purpose of this pull request
close #6632

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by liunaijie** [comment]
I noticed another thing:
[KafkaSinkFactory ](https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/sink/KafkaSinkFactory.java#L45) has format option check, even the `debezium_json` not in expect values but still can pass the check
image

File: docs/en/connector-v2/sink/Kafka.md [hunk 1]
```diff
@@ -30,19 +30,19 @@ They can be downloaded via install-plugin.sh or from the Maven central repositor
...
-| common-options       |        | No       | -       | Source plugin common parameters, please refer to [Source Common Options](common-options.md) for details                                                                                                                                                                                                                                                                                                                                   |
+|         Name         |  Type  | Required | Default |                                                                                                                                                                                                                     Description                                                                                                                                                                                                                     |
```
Comment (litiliu) [inline docs/en/connector-v2/sink/Kafka.md:33]: Could you help explain that changing those lines in the doc helps to solve the https://github.com/apache/seatunnel/issues/6632?

**Review by litiliu — COMMENTED** [review]
Seems  the `canal-json` is not an  incorrect word, why do we need to change it?

File: docs/en/connector-v2/sink/Kafka.md (hunk 1 above)
Comment (liunaijie) [reply docs/en/connector-v2/sink/Kafka.md:33]: in https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/config/MessageFormat.java enum, it's `debezium_json`  not `debezium-json`. so throw debezium-json is not supported error

**Comment by liunaijie** [comment]
> Seems the `canal-json` is not an incorrect word, why do we need to change it?

https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/config/MessageFormat.java#L23 there has `canal_json` format

**Comment by litiliu** [comment]
> > Seems the `canal-json` is not an incorrect word, why do we need to change it?
> 
> https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/config/MessageFormat.java#L23 there has `canal_json` format

is it better to edit the Enum class `MessageFormat` to keep the original names, while setting the value to `canal-json`, and compare the value later?  like this, CANAL_JSON("canal-json")

**Comment by liunaijie** [comment]
> > > Seems the `canal-json` is not an incorrect word, why do we need to change it?
> > 
> > 
> > https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/config/MessageFormat.java#L23 there has `canal_json` format
> 
> is it better to edit the Enum class `MessageFormat` to keep the original names, while setting the value to `canal-json`, and compare the value later? like this, CANAL_JSON("canal-json")

this is the document issue, you can see the demo config from the canal-json document and debezium-json document. they are 
`debezium_json` and `canal_json` instead of  `debezium-json`.



https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/formats/debezium-json.md?plain=1#L91
https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/formats/canal-json.md?plain=1#L98

**Review by litiliu — APPROVED** [review]
Looks good

**Review by EricJoy2048 — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Pull request merged**